core/gpslocation.cpp

```diff
@@ -137,7 +137,9 @@ QString GpsLocation::currentPosition()
 		if (delta < 300) {
 			// we can simply use the last position that we tracked
 			gpsTracker gt = m_trackers.last();
-			QString gpsString = printGPSCoords(&gt.location);
+			char *gps = printGPSCoords(&gt.location);
+			QString gpsString = gps;
+			free(gps);
```

Comment:
we have this pattern three times - I guess it's worth while to have a little C++ wrapper that does this. But that could easily be in a later commit.